roles/repo_setup/tasks/main.yml

```diff
@@ -20,7 +20,8 @@
   ansible.builtin.import_tasks: configure.yml
 - name: Include component repo if required
   ansible.builtin.import_tasks: component_repo.yml
-  when: cifmw_repo_setup_component_name|length > 0
+  when: ( cifmw_repo_setup_component_name|length > 0 and update_playbook_run is undefined and not cifmw_run_update|default(false)) or
+        ( cifmw_repo_setup_component_name|length > 0 and update_playbook_run is defined and cifmw_run_update|default(false))
```

Comment:
i find it hard to follow here.
it seems the conditionals are negating each other

if update is defined and run update

or 

if update is undefined and not run update ?

Reply:
Two conditions are there, because in update CI jobs we have both deployment phase and update phase. When deployment is run in update job cifmw_run_update variable set in zuul globally for deployment and update playbook makes sure this tasks are skipped.
On the other hand, when job with just deployment is started  cifmw_run_update is not set and defaults to false and tasked are imported.